jerry-core/ecma/base/ecma-helpers-value.c

```diff
@@ -615,6 +615,27 @@ ecma_copy_value (ecma_value_t value)  /**< value description */
 } /* ecma_copy_value */
 
 /**
+ * Copy ecma value.
+ *
+ * Note:
+ *   this function is similar to ecma_copy_value, but it is
+ *   faster for direct values since no function call is performed.
+ *   It also increases the binary size so it is recommended for
+ *   critical code paths only.
+ *
+ * @return copy of the given value
+ */
+inline ecma_value_t __attr_always_inline___
+ecma_fast_copy_value (ecma_value_t value)  /**< value description */
+{
+  if (ecma_get_value_type_field (value) != ECMA_TYPE_DIRECT)
+  {
+    value = ecma_copy_value (value);
+  }
+  return value;
```

Comment:
I was hoping this helps for the compiler. Today I analysed a code where the compiler pushed a value to the memory and read it back because it does not understand such a simple construct.